driver/csiplugin/controllerserver.go

```diff
@@ -1600,69 +1605,65 @@ func (cs *ScaleControllerServer) validateSnapId(ctx context.Context, scaleVol *s
+
+func (cs *ScaleControllerServer) validateShallowCopyVolume(ctx context.Context, sourcesnapshot *scaleSnapId, newvolume *scaleVolume) error {
 	loggerId := utils.GetLoggerId(ctx)
 
 	if !newvolume.IsFilesetBased {
-		klog.Errorf("[%s] shallow copy volume as directory based volume is not supported", loggerId)
-		return false
+		klog.Errorf("[%s] creating shallow copy volume as directory based volume is not supported", loggerId)
+		return status.Error(codes.Internal, fmt.Sprintf("creating shallow copy volume as directory based volume is not supported"))
 	}
 
 	if newvolume.ClusterId != sourcesnapshot.ClusterId {
 		klog.Errorf("[%s] shallow copy volume across clusters is not supported", loggerId)
-		return false
+		return status.Error(codes.Internal, fmt.Sprintf("shallow copy volume across clusters is not supported"))
 	}
 
 	if len(newvolume.StorageClassType) != 0 || len(sourcesnapshot.StorageClassType) != 0 {
 		if newvolume.StorageClassType != sourcesnapshot.StorageClassType {
 			klog.Errorf("[%s] validation of shallow copy volume [%s] failed as storage class type is different from source pvc [%s]", loggerId, newvolume.VolName, sourcesnapshot.SnapName)
-			return false
-		} else {
+			return status.Error(codes.Internal, fmt.Sprintf("validation of shallow copy volume [%s] failed as storage class type is different from source pvc [%s]", newvolume.VolName, sourcesnapshot.SnapName))
+		}else{
 			if newvolume.VolBackendFs != sourcesnapshot.FsName {
-				klog.Errorf("[%s] validation of shallow copy volume [%s] failed as filesystem [%s] is different from source pvc [%s] failed ", loggerId, newvolume.VolName,
-					newvolume.VolBackendFs, sourcesnapshot.SnapName)
-				return false
-			} else {
+				klog.Errorf("[%s] validation of shallow copy volume [%s] failed as filesystem [%s] is different from source pvc [%s] failed ", loggerId, newvolume.VolName, 
+				newvolume.VolBackendFs, sourcesnapshot.SnapName)
+				return status.Error(codes.Internal, fmt.Sprintf("validation of shallow copy volume [%s] failed as filesystem [%s] is different from source pvc [%s] failed", newvolume.VolName, newvolume.VolBackendFs, sourcesnapshot.SnapName))
+			}else{
 				if sourcesnapshot.StorageClassType == STORAGECLASS_CLASSIC {
-					isSamefsetType := false
-					if newvolume.FilesetType == independentFileset {
-						if sourcesnapshot.VolType == FILE_INDEPENDENTFILESET_VOLUME {
-							isSamefsetType = true
-						}
-					} else if newvolume.FilesetType == dependentFileset {
-						if sourcesnapshot.VolType == FILE_DEPENDENTFILESET_VOLUME {
-							isSamefsetType = true
-						}
-					}
-
-					if !isSamefsetType {
-						klog.Errorf("[%s] Filesettype is not same for both source snapshot and new volume", loggerId)
-						return false
-					}
-				}
+          				isSamefsetType := false
+          				if newvolume.FilesetType == independentFileset {
```

Comment:
Can we combine this like?
( newvolume.FilesetType == dependentFileset && sourcesnapshot.VolType == FILE_DEPENDENTFILESET_VOLUME ) || ( newvolume.FilesetType == independentFileset && sourcesnapshot.VolType == FILE_INDEPENDENTFILESET_VOLUME)